### PR TITLE
Vm concern handle of migration analytics out-of-sync revision case

### PR DESCRIPTION
**Pull request by gildub**
Resolves https://github.com/konveyor/virt-ui/issues/238

### CODE REVIEW

**Comment by mturley** [comment]
@gildub , based on a discussion with @vconzola I think we actually want this to use the spinner icon (loading status) with the text "Analyzing" instead of unknown. Also, we should add a condition for this in the `VMConcernsDescription` that displays 
"Completing migration analysis. This might take a few minutes." and hides the list of concerns (since it will be out of date in this state).

File: src/app/Plans/components/Wizard/VMConcernsIcon.tsx [hunk 1]
```diff
@@ -2,15 +2,28 @@ import * as React from 'react';
...
 interface IVMConcernsIconProps {
   vm: IVMwareVM;
 }
 
 const VMConcernsIcon: React.FunctionComponent<IVMConcernsIconProps> = ({
   vm,
 }: IVMConcernsIconProps) => {
+  if (vm.revisionAnalyzed < vm.revision) {
+    return (
+      <Flex spaceItems={{ default: 'spaceItemsSm' }}>
```
Comment (mturley) [inline src/app/Plans/components/Wizard/VMConcernsIcon.tsx:18]: Can't we just use a `<StatusIcon status={StatusType.Loading}>` here?

File: src/app/Plans/components/Wizard/VMConcernsIcon.tsx [hunk 2]
```diff
@@ -2,15 +2,28 @@ import * as React from 'react';
...
+        <FlexItem>
+          <Spinner aria-valuetext="Analysing" size="sm" className="spinner" />
+        </FlexItem>
+        <FlexItem>Analysing</FlexItem>
```
Comment (mturley) [inline src/app/Plans/components/Wizard/VMConcernsIcon.tsx:22]: @vconzola should we use "Analysing" or "Analyzing" with a z? I think with a z is the American spelling and everywhere else in the world spells it with an s.

File: src/app/Providers/components/CloudAnalyticsInfoAlert.tsx [hunk 3]
```diff
@@ -9,7 +9,11 @@ const CloudAnalyticsInfoAlert: React.FunctionComponent = () => {
     LocalStorageKey.isProvidersPageMAAlertHidden
   );
 
-  const link = <a href={CLOUD_MA_LINK.href}>{CLOUD_MA_LINK.label}</a>;
```
Comment (mturley) [inline src/app/Providers/components/CloudAnalyticsInfoAlert.tsx:12]: Looks like this change carried over from your other PR by accident, need to rebase.

File: src/app/queries/types/vms.types.ts [hunk 4]
```diff
@@ -7,11 +7,12 @@ export interface IVMwareVMDisk {
 
 export interface IVMwareVMConcern {
   name: string;
-  severity: 'Warning' | 'Critical' | 'Advisory' | 'Info'; // TODO not sure if these are final
+  severity: 'Warning' | 'Critical' | 'Advisory' | 'Info' | 'Unknown';
```
Comment (mturley) [inline src/app/queries/types/vms.types.ts:10]: I don't think `'Unknown'` is a valid severity value in the API, but I'm not sure. We're not handling it above, either way, should probably remove it

**Comment by vconzola** [comment]
Use the American English spelling (just like we use Canceled instead of Cancelled). Localization can change/fix it if necessary.

File: src/app/Plans/components/Wizard/VMConcernsIcon.tsx (hunk 1 above)
Comment (gildub) [reply src/app/Plans/components/Wizard/VMConcernsIcon.tsx:18]: Of course we can!

**Review by mturley — APPROVED** [review]

**Pull request merged**